src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting/DefaultRazorFormattingServiceTest.cs

```diff
@@ -0,0 +1,383 @@
+}
+}|
+",
+expected: @"
+@functions {
+ public class Foo{
+void Method() { <div></div> }
+}
+}
+");
+        }
+
+        [Fact]
+        public async Task DoesNotFormat_CodeBlockDirectiveWithImplicitExpressions()
+        {
+            await RunFormattingTestAsync(
+input: @"
+|@functions {
+ public class Foo{
+void Method() { @DateTime.Now }
+}
+}|
+",
+expected: @"
+@functions {
+ public class Foo{
+void Method() { @DateTime.Now }
+}
+}
+");
+        }
+
+        [Fact]
+        public async Task DoesNotFormat_CodeBlockDirectiveWithExplicitExpressions()
+        {
+            await RunFormattingTestAsync(
+input: @"
+|@functions {
+ public class Foo{
+void Method() { @(DateTime.Now) }
+}
+}|
+",
+expected: @"
+@functions {
+ public class Foo{
+void Method() { @(DateTime.Now) }
+}
+}
+");
+        }
+
+        [Fact]
+        public async Task DoesNotFormat_CodeBlockDirectiveWithRazorComments()
+        {
+            await RunFormattingTestAsync(
+input: @"
+|@functions {
+ public class Foo{
+@* This is a Razor Comment *@
+void Method() {  }
+}
+}|
+",
+expected: @"
+@functions {
+ public class Foo{
+@* This is a Razor Comment *@
+void Method() {  }
+}
+}
+");
+        }
+
+        [Fact]
+        public async Task DoesNotFormat_CodeBlockDirectiveWithRazorStatements()
+        {
+            await RunFormattingTestAsync(
+input: @"
+|@functions {
+ public class Foo{
+@* This is a Razor Comment *@
+void Method() { @if (true) {} }
+}
+}|
+",
+expected: @"
+@functions {
+ public class Foo{
+@* This is a Razor Comment *@
+void Method() { @if (true) {} }
+}
+}
+");
+        }
+
+        [Fact]
+        public async Task OnlyFormatsWithinRange()
+        {
+            await RunFormattingTestAsync(
+input: @"
+@functions {
+ public class Foo{}
+        |public interface Bar {
+}|
+}
+",
+expected: @"
+@functions {
+ public class Foo{}
+    public interface Bar
+    {
+    }
+}
+");
+        }
+
+        [Fact]
+        public async Task MultipleCodeBlockDirectives()
+        {
+            await RunFormattingTestAsync(
+input: @"
+|@functions {
+ public class Foo{}
+        public interface Bar {
+}
+}
+Hello World
+@functions {
+      public class Baz    {
+          void Method ( )
+          { }
+          }
+}|
+",
+expected: @"
+@functions {
+    public class Foo { }
+    public interface Bar
+    {
+    }
+}
+Hello World
+@functions {
+    public class Baz
+    {
+        void Method()
+        { }
+    }
+}
+");
+        }
+
+        [Fact]
+        public async Task MultipleCodeBlockDirectives2()
+        {
+            await RunFormattingTestAsync(
+input: @"|
+Hello World
+@functions {
+public class HelloWorld
+{
+}
+}
+
+@functions{
+    
+ public class Bar {}
+}
+|",
+expected: @"
+Hello World
+@functions {
+    public class HelloWorld
+    {
+    }
+}
+
+@functions{
+    
+    public class Bar { }
+}
+");
+        }
+
+        [Fact]
+        public async Task CodeOnTheSameLineAsCodeBlockDirectiveStart()
```

Comment:
What happens with:

```Razor
Hello World    @functions {
...
}
```

Reply:
Nothing. It doesn't touch the block start if there is non-whitespace preceding it.
```razor
Hello World    @functions {
    ...
}
```
I'll add a test